scripts/zkai_stwo_ai_boundary_pressure_inventory_gate.py

```
@@ -0,0 +1,605 @@
+    writer.writeheader()
+    writer.writerows(rows)
+    return out.getvalue()
+
+
+def reject_bad_output_path(path: pathlib.Path) -> None:
+    resolved = path.resolve()
+    evidence = EVIDENCE_DIR.resolve()
+    try:
+        resolved.parent.relative_to(evidence)
+    except ValueError as err:
+        raise StwoAiBoundaryPressureInventoryError("output path must stay inside evidence dir") from err
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)